csrc/dynamic_transform.cpp

```diff
@@ -900,6 +900,8 @@ void DynamicTransformConcretizer::concretizeReshape() {
     // would fail at this point. So we skip checkConcretizedUses here and
     // perform it later in mutate(TensorView*).
 
+    registerMutation(incomplete_out_tv, concrete_reshape_out_tv);
```

Comment:
Can we add a comment here and in the other case that we are adding this call specifically so that the replacement is reflected in mutations_ and that this call does not mean the tv will be automatically replaced (we manually replace it here instead).